File: 1217-relative-sort-array/1217-relative-sort-array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        map<int,int>mp;
        vector<int>res;
        for(int i=0;i<arr1.size();i++){
            mp[arr1[i]]++;
        }
        for(int i=0;i<arr2.size();i++){
            while(mp[arr2[i]]>0){
                res.push_back(arr2[i]);
                mp[arr2[i]]--;
            }
            mp.erase(arr2[i]);
        }
        vector<int>temp;
        for(auto it=mp.begin();it!=mp.end();it++){
            while(it->second>0){
                temp.push_back(it->first);
                it->second--;
            }
            mp.erase(it->first);
        }
        sort(temp.begin(), temp.end());
        res.insert(res.end(), temp.begin(), temp.end());
        return res;
    }
};

File: 1217-relative-sort-array/1217-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int i1 = 0, i2 = 0;
        int n = arr2.length;
        while(i2 < n){
            i1 = swap(arr1, arr2[i2++], i1);
        }
        Arrays.sort(arr1, i1, arr1.length);
        return arr1;
    }
    int swap(int[] arr, int target, int from){
        int i = from;
        while(i < arr.length){
            if(arr[i] == target){
                int temp = arr[from];
                arr[from++] =  arr[i];
                arr[i] = temp;
            }
            i++;
        }
        return from;
    }
}